src/apps/migrate/services/postgres.py

```diff
@@ -310,3 +310,30 @@ def save_user_pins(self, user_pin_dao):
         """
         rows_count = self.insert_dao_collection(sql, list(user_pin_dao))
         migration_log.warning(f"Inserted {rows_count} rows")
+
+    def get_migrated_workspaces(self) -> List[uuid.UUID]:
+        sql = "SELECT id FROM users_workspaces"
+        return self.get_pk_array(sql, as_bson=False)
+
+    def execute_in_transact(self, dao_list, err_prefix=""):
+        success, skip = 0, 0
+        for dao in dao_list:
+            try:
+                cursor = self.connection.cursor()
+                sql = str(dao)
+                cursor.execute(sql)
+                success += 1
+            except Exception as ex:
+                skip += 1
+                if not hasattr(ex, "pgcode"):
+                    # not pg error
+                    raise ex
+                if getattr(ex, "pgcode") == FOREIGN_KEY_VIOLATION:
+                    migration_log.warning(f"[{err_prefix}] {ex}")
+                elif getattr(ex, "pgcode") == UNIQUE_VIOLATION:
+                    migration_log.warning(f"[{err_prefix}] {ex}")
```

Comment:
This is necessary so that the migration does not fall if data was inserted into the table during the of applets migration . So yes, I have to say that it is a must to have in this case

Reply:
Here it is mandatory. we don't know which users were migrated in advance, it's easier to catch it here than to carry a list of existing ones every time